### PR TITLE
use builder to specify custom name prefix instead

### PR DESCRIPTION
**Pull request by BusyJay**
So the old API won't break and it will be more convenient to add extra configuration in the future.

### CODE REVIEW

File: src/env.rs [hunk 1]
```diff
@@ -38,31 +38,44 @@ fn poll_queue(cq: Arc<CompletionQueue>) {
...
 }
 
-/// An object that used to control concurrency and start event loop.
-pub struct Environment {
-    cqs: Vec<Arc<CompletionQueue>>,
-    idx: AtomicUsize,
-    _handles: Vec<JoinHandle<()>>,
+pub struct EnvBuilder {
+    cq_count: usize,
+    name_prefix: Option<String>,
```
Comment (siddontang) [inline src/env.rs]: I think we can use String directly with name "grpc-poll".

File: src/env.rs [hunk 2]
```diff
@@ -73,6 +86,26 @@ impl Environment {
...
+    cqs: Vec<Arc<CompletionQueue>>,
+    idx: AtomicUsize,
+    _handles: Vec<JoinHandle<()>>,
+}
+
+impl Environment {
+    /// Initialize grpc and create a threadpool to poll event loop.
+    ///
+    /// Each thread in threadpool will have one event loop.
+    pub fn new(cq_count: usize) -> Environment {
```
Comment (siddontang) [inline src/env.rs]: Seem we can remove the construct function, only use Builder outside.

Comment (BusyJay) [inline src/env.rs]: It's just a shortcut to create `Environment`.

File: src/env.rs (hunk 1 above)
Comment (BusyJay) [inline src/env.rs]: What if caller dosen't want any custom name?

**Pull request merged**